design/design-generic-references.md

```diff
@@ -0,0 +1,477 @@
+# Generic Resource References
+
+* Owner: Predrag Knezevic (@pedjak)
+* Reviewers: Crossplane Maintainers
```

Comment:
Let's nominate two specific Crossplane maintainers to make it clear who we expect to approve this design.

Reply:
@crossplane/crossplane-maintainers please your hand :)

Reply:
Count me in